addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -462,18 +466,30 @@ export async function activateCropper(image, aspectRatio, dataset) {
     return new Promise(resolve => image.addEventListener('ready', resolve, {once: true}));
 }
 /**
- * Marks an <img> with its attachment data (originalId, originalSrc, mimetype)
+ * Loads the image options (filter, shape etc...)
  *
  * @param {HTMLImageElement} img the image whose attachment data should be found
+ * @param {Object} recordInfo a dictionary of the information linked to the
+ * closest editable element of the image.
  * @param {string} [attachmentSrc=''] specifies the URL of the corresponding
  * attachment if it can't be found in the 'src' attribute.
  */
-export async function loadImageInfo(img, attachmentSrc = '') {
-    const src = attachmentSrc || img.getAttribute('src');
-    // If there is a marked originalSrc, the data is already loaded.
-    // If the image does not have the "mimetypeBeforeConversion" attribute, it
-    // has to be added.
-    if ((img.dataset.originalSrc && img.dataset.mimetypeBeforeConversion) || !src) {
+export async function loadImageInfo(img, recordInfo, attachmentSrc = "") {
+    const snakeToCamelCase = (dictionary) => {
+        // Convert all the keys of a dictionary from snake to camel case
+        const camelEntryDic = {};
+        const snakeEntries = Object.keys(dictionary);
```

Comment:
IMO, those should be named `camelDict` (it's just a dictionary, not a dictionary where e.g. keys lead to entries (aka key + value)), and `snakeKeys` because it's just the list of keys.
(and inside the lambda: `snakeKey` and `camelKey`)